pkg/kubelet/prober/worker.go

```diff
@@ -330,6 +341,38 @@ func (w *worker) doProbe(ctx context.Context) (keepGoing bool) {
 	return true
 }
 
+func (w *worker) getOrSetHttpReq(status v1.PodStatus) (*http.Request, error) {
+	probeSpec, err := getProbeSpec(&w.container, w.probeType)
+	if err != nil {
+		return nil, err
+	}
+
+	// create http request object for HTTP probes only.
+	if probeSpec.HTTPGet != nil {
+		// only create request object if it doesn't exist.
+		if w.httpReq == nil {
+			req, err := httpprobe.NewRequestForHTTPGetAction(probeSpec.HTTPGet, &w.container, status.PodIP, "probe")
+			if err != nil {
+				return nil, err
+			}
+
+			// persist request for reuse in future.
```

Comment:
is it valid to reuse a request and context multiple times?

Reply:
Also, I was thinking about this but I'm not sure, the request is build from some pod/containers parameters ` pod *v1.Pod, status v1.PodStatus, container v1.Container, containerID kubecontainer.ContainerID, req *http.Request)`, ... are these parameters guaranteed to be constant? 

pods can change the IP,  at least Pod.Status.PodIP. is not immutable ... I'm not familiar enough with the pod lifecycle to know if this can cause an issue here too

Reply:
it is valid from golang perspective for our basic use case. Here is some discussion: https://github.com/golang/go/issues/19653.

Changing PodIP is an interesting possibility. Need to look deeper. But even if it is the case we can just override the Host on the same request object instead of creating a brand new one. Probe calls are sequencial so shouldn't be a problem